Release v3.5.1 (#60)
DeterminateSystems/nix-src: doc/manual/source/release-notes-determinate/changes.md

<!-- Determinate Nix version 3.4.2 -->

<!-- Determinate Nix version 3.5.0 -->

<!-- Determinate Nix version 3.5.1 -->

> * Deprecate upgrade-nix command by @gustavderdrache in [DeterminateSystems/nix-src#55](https://github.com/DeterminateSystems/nix-src/pull/55)

Reviewer: ```suggestion
* Deprecate upgrade-nix command in favor of `determinate-nixd upgrade`, by @gustavderdrache in [DeterminateSystems/nix-src#55](https://github.com/DeterminateSystems/nix-src/pull/55)
```

* Lazy trees v2 by @edolstra in [DeterminateSystems/nix-src#27](https://github.com/DeterminateSystems/nix-src/pull/27)

* Improve lazy trees backward compatibility by @edolstra in [DeterminateSystems/nix-src#56](https://github.com/DeterminateSystems/nix-src/pull/56)

* Canonicalize flake input URLs before checking flake.lock file staleness, for dealing with `dir` in URL-style flakerefs by @edolstra in [DeterminateSystems/nix-src#57](https://github.com/DeterminateSystems/nix-src/pull/57)

> * Improve build failure error messages by @edolstra in [DeterminateSystems/nix-src#58](https://github.com/DeterminateSystems/nix-src/pull/58)

Reviewer: ```suggestion
* Improved the build failure and dependency failure error messages to include output paths, by @edolstra in [DeterminateSystems/nix-src#58](https://github.com/DeterminateSystems/nix-src/pull/58)
```

<!-- Determinate Nix version 3.4.2 -->

<!-- Determinate Nix version 3.5.0 -->

<!-- Determinate Nix version 3.5.1 -->

> 
> * Only run the test step after build completes by @grahamc in [DeterminateSystems/nix-src#51](https://github.com/DeterminateSystems/nix-src/pull/51)
> 
> * Remove flake-compat input by @lucperkins in [DeterminateSystems/nix-src#52](https://github.com/DeterminateSystems/nix-src/pull/52)

Reviewer: ```suggestion
```

* Deprecate upgrade-nix command by @gustavderdrache in [DeterminateSystems/nix-src#55](https://github.com/DeterminateSystems/nix-src/pull/55)

> 
> * Update flake.lock in light of recent change by @lucperkins in [DeterminateSystems/nix-src#54](https://github.com/DeterminateSystems/nix-src/pull/54)

Reviewer: ```suggestion
```

* Lazy trees v2 by @edolstra in [DeterminateSystems/nix-src#27](https://github.com/DeterminateSystems/nix-src/pull/27)

* Improve lazy trees backward compatibility by @edolstra in [DeterminateSystems/nix-src#56](https://github.com/DeterminateSystems/nix-src/pull/56)

* Canonicalize flake input URLs before checking flake.lock file staleness, for dealing with `dir` in URL-style flakerefs by @edolstra in [DeterminateSystems/nix-src#57](https://github.com/DeterminateSystems/nix-src/pull/57)

* Improve build failure error messages by @edolstra in [DeterminateSystems/nix-src#58](https://github.com/DeterminateSystems/nix-src/pull/58)

> * Release v3.5.0 by @github-actions in [DeterminateSystems/nix-src#59](https://github.com/DeterminateSystems/nix-src/pull/59)
> 
> 
> * @gustavderdrache made their first contribution in [DeterminateSystems/nix-src#55](https://github.com/DeterminateSystems/nix-src/pull/55)
> 
> * Dispatch release notes with a gh token by @grahamc in [DeterminateSystems/nix-src#61](https://github.com/DeterminateSystems/nix-src/pull/61)

Reviewer: ```suggestion
```

<!-- Determinate Nix version 3.4.2 -->

<!-- Determinate Nix version 3.5.0 -->

<!-- Determinate Nix version 3.5.1 -->

> * Dispatch release notes with a gh token by @grahamc in [DeterminateSystems/nix-src#61](https://github.com/DeterminateSystems/nix-src/pull/61)

Reviewer: ```suggestion
* Only run the test step after build completes by @grahamc in [DeterminateSystems/nix-src#51](https://github.com/DeterminateSystems/nix-src/pull/51)

* Remove flake-compat input by @lucperkins in [DeterminateSystems/nix-src#52](https://github.com/DeterminateSystems/nix-src/pull/52)

* Deprecate upgrade-nix command by @gustavderdrache in [DeterminateSystems/nix-src#55](https://github.com/DeterminateSystems/nix-src/pull/55)

* Update flake.lock in light of recent change by @lucperkins in [DeterminateSystems/nix-src#54](https://github.com/DeterminateSystems/nix-src/pull/54)

* Lazy trees v2 by @edolstra in [DeterminateSystems/nix-src#27](https://github.com/DeterminateSystems/nix-src/pull/27)

* Improve lazy trees backward compatibility by @edolstra in [DeterminateSystems/nix-src#56](https://github.com/DeterminateSystems/nix-src/pull/56)

* Canonicalize flake input URLs before checking flake.lock file staleness, for dealing with `dir` in URL-style flakerefs by @edolstra in [DeterminateSystems/nix-src#57](https://github.com/DeterminateSystems/nix-src/pull/57)

* Improve build failure error messages by @edolstra in [DeterminateSystems/nix-src#58](https://github.com/DeterminateSystems/nix-src/pull/58)

* Release v3.5.0 by @github-actions in [DeterminateSystems/nix-src#59](https://github.com/DeterminateSystems/nix-src/pull/59)

* @gustavderdrache made their first contribution in [DeterminateSystems/nix-src#55](https://github.com/DeterminateSystems/nix-src/pull/55)

* Dispatch release notes with a gh token by @grahamc in [DeterminateSystems/nix-src#61](https://github.com/DeterminateSystems/nix-src/pull/61)
```
